Panic calling domainName

**Describe the bug**
Panic calling domainName from shell.

**To Reproduce**

```
benr@swell:~$ cnquery version
cnquery 7.17.2 (cdfdda91, 2023-02-16T01:01:16Z)
benr@swell:~$ cnquery shell
→ no provider specified, defaulting to local.
  Use --help for a list of available providers.
→ loaded configuration from /home/benr/.config/mondoo/mondoo-cuddletech.yml using source $MONDOO_CONFIG_PATH
→ discover related assets for 1 asset(s)
→ resolved assets resolved-assets=1
  ___ _ __   __ _ _   _  ___ _ __ _   _ 
 / __| '_ \ / _` | | | |/ _ \ '__| | | |
| (__| | | | (_| | |_| |  __/ |  | |_| |
 \___|_| |_|\__, |\__,_|\___|_|   \__, |
  mondoo™      |_|                |___/  interactive shell

cnquery> domainName
panic: interface conversion: interface {} is nil, not string

goroutine 52 [running]:
go.mondoo.com/cnquery/resources/packs/core.(*mqlDomainName).init(0xc0006a6288, 0xc0006a6280)
        /home/benr/actions-runner/_work/cnquery/cnquery/resources/packs/core/dns.go:32 +0x371
go.mondoo.com/cnquery/resources/packs/core.newDomainName(0xc000f5bc40, 0x0?)
        /home/benr/actions-runner/_work/cnquery/cnquery/resources/packs/core/core.lr.go:10125 +0xcf
go.mondoo.com/cnquery/resources.(*Runtime).CreateResourceWithID(0xc000f5bc40, {0xc0008ca760?, 0x0?}, {0x0, 0x0}, {0xe267f20, 0x0, 0x0})
        /home/benr/actions-runner/_work/cnquery/cnquery/resources/runtime.go:192 +0x34c
go.mondoo.com/cnquery/resources.(*Runtime).CreateResource(...)
        /home/benr/actions-runner/_work/cnquery/cnquery/resources/runtime.go:223
go.mondoo.com/cnquery/llx.(*blockExecutor).createResource(0xc0002a7180, {0xc0008ca760, 0xa}, 0xa?, 0xc0008cabd8?, 0x5dadc60?)
        /home/benr/actions-runner/_work/cnquery/cnquery/llx/llx.go:679 +0x12c
go.mondoo.com/cnquery/llx.(*blockExecutor).runGlobalFunction(0xc0002a7180, 0xc00043ec30, 0xe0ab720, 0xa?)
        /home/benr/actions-runner/_work/cnquery/cnquery/llx/llx.go:724 +0x2af
go.mondoo.com/cnquery/llx.(*blockExecutor).runFunction(0xc0002a7180?, 0xc00043ef00?, 0x203000?)
        /home/benr/actions-runner/_work/cnquery/cnquery/llx/llx.go:750 +0xfd
go.mondoo.com/cnquery/llx.(*blockExecutor).runChunk(0xc0002a7180, 0x5dade2e?, 0x100000001?)
        /home/benr/actions-runner/_work/cnquery/cnquery/llx/llx.go:781 +0x25e
go.mondoo.com/cnquery/llx.(*blockExecutor).runRef(0xc0008f1620?, 0x6?)
        /home/benr/actions-runner/_work/cnquery/cnquery/llx/llx.go:806 +0xef
go.mondoo.com/cnquery/llx.(*blockExecutor).runChain(0xc0002a7180, 0x8e07f4c?)
        /home/benr/actions-runner/_work/cnquery/cnquery/llx/llx.go:838 +0xa6
go.mondoo.com/cnquery/llx.(*blockExecutor).run(0xc0002a7180)
        /home/benr/actions-runner/_work/cnquery/cnquery/llx/llx.go:351 +0x2d7
go.mondoo.com/cnquery/llx.(*MQLExecutorV2).Run(0xc001518300?)
        /home/benr/actions-runner/_work/cnquery/cnquery/llx/llx.go:283 +0x59
go.mondoo.com/cnquery/mql/internal.(*executionManager).executeCodeBundle(0xc00043ef50, 0xc0005b43c0, 0x406f51?, {0x0, 0x0})
        /home/benr/actions-runner/_work/cnquery/cnquery/mql/internal/execution_manager.go:167 +0x35d
go.mondoo.com/cnquery/mql/internal.(*executionManager).Start.func1()
        /home/benr/actions-runner/_work/cnquery/cnquery/mql/internal/execution_manager.go:85 +0x1cb
created by go.mondoo.com/cnquery/mql/internal.(*executionManager).Start
        /home/benr/actions-runner/_work/cnquery/cnquery/mql/internal/execution_manager.go:54 +0x6a

```

--- a/resources/packs/core/dns.go
+++ b/resources/packs/core/dns.go
@@ -29,15 +29,17 @@ func (d *mqlDomainName) init(args *resources.Args) (*resources.Args, DomainName,
 		(*args)["fqdn"] = fqdn
 	}
 
-	dn, err := domain.Parse(fqdn.(string))
-	if err != nil {
-		return nil, nil, err
-	}
+	if fqdn != nil {
+		dn, err := domain.Parse(fqdn.(string))
+		if err != nil {
+			return nil, nil, err
+		}
 
-	(*args)["effectiveTLDPlusOne"] = dn.EffectiveTLDPlusOne
-	(*args)["tld"] = dn.TLD
-	(*args)["tldIcannManaged"] = dn.IcannManagedTLD
-	(*args)["labels"] = StrSliceToInterface(dn.Labels)
+		(*args)["effectiveTLDPlusOne"] = dn.EffectiveTLDPlusOne
+		(*args)["tld"] = dn.TLD
+		(*args)["tldIcannManaged"] = dn.IcannManagedTLD
+		(*args)["labels"] = StrSliceToInterface(dn.Labels)
+	}
 
 	return args, nil, nil
 }

--- a/resources/packs/core/dns.go
+++ b/resources/packs/core/dns.go
@@ -29,17 +29,19 @@ func (d *mqlDomainName) init(args *resources.Args) (*resources.Args, DomainName,
 		(*args)["fqdn"] = fqdn
 	}
 
-	if fqdn != nil {
-		dn, err := domain.Parse(fqdn.(string))
-		if err != nil {
-			return nil, nil, err
-		}
-
-		(*args)["effectiveTLDPlusOne"] = dn.EffectiveTLDPlusOne
-		(*args)["tld"] = dn.TLD
-		(*args)["tldIcannManaged"] = dn.IcannManagedTLD
-		(*args)["labels"] = StrSliceToInterface(dn.Labels)
+	if fqdn == nil {
+		return nil, nil, errors.New("domainName resource requires fqdn argument")
 	}
+
+	dn, err := domain.Parse(fqdn.(string))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	(*args)["effectiveTLDPlusOne"] = dn.EffectiveTLDPlusOne
+	(*args)["tld"] = dn.TLD
+	(*args)["tldIcannManaged"] = dn.IcannManagedTLD
+	(*args)["labels"] = StrSliceToInterface(dn.Labels)
 
 	return args, nil, nil
 }

--- a/resources/packs/core/dns_test.go
+++ b/resources/packs/core/dns_test.go
@@ -10,3 +10,10 @@ func TestResource_DNS(t *testing.T) {
 	res := x.TestQuery(t, "dns(\"mondoo.com\").mx")
 	assert.NotEmpty(t, res)
 }
+
+func TestResource_DomainName(t *testing.T) {
+	res := x.TestQuery(t, "domainName")
+	assert.NotEmpty(t, res)
+	res = x.TestQuery(t, "domainName(\"mondoo.com\").tld")
+	assert.Equal(t, "com", string(res[0].Result().Data.Value))
+}

